### DISCUSSION TITLE
create rules / whitelisting for suricata

### DISCUSSION THREAD

**Discussion by Thomer73** [Security-Onion-Solutions/securityonion#5315] [Unsupported Versions] [answered]
Hello,

we're trying to build some pass-action rules in the local.rules to whitelist some alerts. These should skip detect by other sid patterns. My problem, the local.rules will merged to all.rules (suricata) to bottom of the file. So the "pattern" rules will detect traffic, before the own pass rules get in action. We tried to use lower sid numbers and hope this will put the own rules to the top of all.rules.
But it will not work. Is there a way to change this?
-------------------------
The **suricata.yaml** have this section:

> default-rule-path: /etc/suricata/rules
> rule-files:
> - all.rules

I don't know how to put another, for example ignore.rules before the all.rules entry. My hope is that the rules in this files will prior by the system. Is it possible to put the enty in /opt/so/saltstack/local/pillar/global.sls ? the suricata.yaml will be overwritten with info from the global.sls , or?

Thanks so much
Thomas

**Comment by weslambert** [comment, marked as answer]
TMK, the order within the file has nothing to do with how the rules are evaluated.  You'll probably want to look into `action-order`, and add that to the pillar for suricata:

https://suricata.readthedocs.io/en/suricata-6.0.0/configuration/suricata-yaml.html#action-order

**Reply by Thomer73** [reply]
Hi, thanks for your answer. 
yesterday i found the action-order and now i know that pass rules will take prior action.
My second problem: some local.rules weren't merged into rules.all. i found a article from snort and user sid have to begin above :999999.
now it works ;-)

Thanks a lot.
thomas
